scalems/wrappers/__init__.py

```diff
@@ -16,32 +15,109 @@
 
+        Proposal: At an abstract level, this contains everything needed to define the simulation 
+        mathematically. The chemical system, instructions for running the simulation, but 
+        not how it's executed, or the resources it will take. That will be taken care of later in
+        the workflow.
+
+        GROMACS: .mpd,.top,.gro
+        OpenMM: (look at ExTasy workflows)
+        PLUMED-GROMACS: GROMACS + PLUMED input file
+        LAMMPS: .input file
+
+        Crucially, it provide a place for the information to be staged
+        for modifying input, such that a single simulation can be
+        loaded, and then variants of that simulation can be "cloned out", and modified.
+
+        For many programs, this may simply be putting the text of the
+        files into a container where they can be retrived and
+        modified.
+
     modify_input(simulation_input: SimulationInput, **kwargs) -> SimulationInput:
+
         A simulation input data source produced by *make_input* or *modify_input*
         may be modified in a pipe line through (implementation dependent) key word
         arguments to a *modify_input* command.
 
+        For most options, then a keyword argument can directly
+        parallel the keyword syntax used by the simulation programs
+        themselves: LAMMPS, PLUMED, GROMACS (in the .mdp) all use
+        keyword options.  It's not as clear for OpenMM, but we should
+        use the ExTASY implementations to guide us as to how modification is done there.  
+
+        One question is how the defintion of the chemical system
+        should changed using this approach. This is not set up very
+        well in the keyword format.  We could, for now, just replace
+        the file containing the molecular system definition entirely
+        using a modify_input.  This might be the best bet for now, but
+        we will want to think about what sorts of changes might be of
+        interest, if any in the future.
+
+        The issue is that in most cases, if the chemical system is
+        changing, it usually requires changes of many lines of
+        topologies, and so passing it in as a dictionary will likely
+        be a big pain. Likely easier to take care of it outside the
+        simulation for now. Eventually, if some parallel work on
+        simulation topology definitions in Open Force Field works,
+        then it may be possible to do this better.
+
+        Things like "training potential" that get attached in the
+        examples should probably be attached at this point in the
+        workflow, rather than at the simulate. For most workflows, it
+        could ONLY be added at this stage, since potentials are
+        defined in the input files.  For GMX_API driven simulations,
+        the actual attachment of the potentials to the md simulation
+        could happen in the simulate function, but it would be better
```

Comment:
I disagree--I think it is best to say the "simulation input" is everything *internal* to the simulation, but *external* elements can potentially interact and alter the simulation. I think it's a bad idea to limit use cases here.

Reply:
@peterkasson I'm not sure what you mean by the distinction external vs. external.  Can you clarify?  By simulation, I mean "Everything required to define the physics of a system".  I would classify the attachment of additional potentials (like in GMXAPI) as being part of the simulation, not being external to the simulation.  Without them, it's a different simulation.